## 1. Allele frequencies and Allele Frequency Spectrum  

Navigate to the folder `1_afs` within the `mytutorial` folder 

```shell
cd mytutorial
cd 1_afs
```

### Overall allele frequencies 
Calculate allele frequencies directly form vcf, only for the snps, all samples 

``` shell
plink2 --vcf /home/genomics/workshop_materials/population_genomics/chr15.pan.fa.a2fb268.4030258.6a1ecc2.smooth.reliable.vcf.gz \
       --freq \
       --allow-extra-chr \
       --vcf-half-call m \
       --snps-only \
       --out chr15

## --vcf loads a genotype VCF file
## --allow-extra-chr  force to accept chromosome code 'chm13#chr15'
## --vcf-half-call specify how GT half-call should be processed. See tutorial section 0 for more info
## --snps-only  selects only loci that are single nucleotide polymorphisms 
##--out gives a human-readable name 

```

This command will produce these files: 

```shell
-rw-rw-r-- 1 enza enza 33492352 gen 14 11:28 chr15.afreq
-rw-rw-r-- 1 enza enza      923 gen 14 11:28 chr15.log
```

Open the `.log` file  and read it to check that all went well. 

Open the `.afreq` file and look at it. Check the [info](https://www.cog-genomics.org/plink/2.0/formats#afreq) on the content of the columns and get familiar with it:  

```shell 
cat chr15.afreq | column -t | less -S 
```
Can you guess why for some loci the allele frequency is null (nan)? 

***
### Population-specific allele frequencies 

We now want to calculate allele frequencies specifically for SNPs within African and American populations. We will use the `plink2` fileset instead of the `vcf`, just to learn another command. 

The crucial first step involves identifying the samples that belong to these specific populations. This information is in metadata files that we need to reference:  

```shell
plink2 --pfile  ../data/chr15 \
       --keep  ../../metadata/afr.onlyid \
       --freq \
       --allow-extra-chr \
       --snps-only \
       --out chr15.afr 

plink2 --pfile  ../data/chr15 \
       --keep  ../../metadata/amr.onlyid \
       --freq \
       --allow-extra-chr \
       --snps-only \
       --out chr15.amr 

## --pfile specifies which plink2 file set to use, you only need to specify the file prefix (e.g. chr15) 
## --keep reference to the file containing the individuals to filter 
```

This will produce: 
```shell
-rw-rw-r-- 1 enza enza 32483032 gen 14 11:38 chr15.afr.afreq
-rw-rw-r-- 1 enza enza      863 gen 14 11:38 chr15.afr.log
-rw-rw-r-- 1 enza enza 30547659 gen 14 11:38 chr15.amr.afreq
-rw-rw-r-- 1 enza enza      863 gen 14 11:38 chr15.amr.log

```

As usual open the log and check for error messages if any. 
Open the `.afreq` files and look at them 
 
***

### Allele Frequncy Spectrum (AFS) 

In plink2, the `--freq` option includes the `refbin` and `alt1bin` modifiers for generating histogram counts during allele frequency calculations. To use these modifiers, you must specify the bin intervals either directly in the command line or by referencing a file, and using the `refbin-file` and `alt1bin-file` modifiers. 

We can produce the bin info file with a simple shell command that write a sequence form 0.1 to 1 in intervals of 0.05 and store it in a file called `mybins.txt`: 
 
```shell
LC_NUMERIC="C" seq -s " " 0.1 0.05 1.0 > mybins.txt
````

Look at the `mybins.txt` file: 

```shell
cat mybins.txt 
```

***
Calculate the counts for the AFS based on alternate allele frequencies:  

```shell
plink2 --pfile  ../data/chr15 \
       --keep  .../../metadata/afr.onlyid \
       --freq alt1bins-file='mybins.txt' \
       --allow-extra-chr --snps-only \
       --out chr15.afr

plink2 --pfile  ../data/chr15 \
       --keep  ../../metadata/amr.onlyid \
       --freq alt1bins-file='mybins.txt' \
       --allow-extra-chr \
       --snps-only \
       -out chr15.amr

## --freq alt1bins-file='mybins.txt' specify to bin the frequency as described in the file mybins.txt 

```

Plot the AFS 

```shell
Rscript  ../rscripts/plotafs.R chr15.afr.afreq.ref.bins chr15.amr.afreq.ref.bins afr-amr-ALT 
```



<!----
***
sfs of the reference allele 
```
plink2 --pfile  ../data/chr15 --keep  ../data/pop/afr.onlyid --freq refbins-file='mybins.txt'  --allow-extra-chr --snps-only   --out chr15.afr
```

```
plink2 --pfile  ../data/chr15 --keep  ../data/pop/amr.onlyid --freq refbins-file='mybins.txt'  --allow-extra-chr --snps-only   --out chr15.amr
```

```
Rscript  ../rscripts/plotafs.R chr15.afr.afreq.alt1.bins chr15.amr.afreq.alt1.bins afr-amr-REF
``` 
> 

