## 1. Allele frequencies and Allele Frequency Spectrum  

Navigate to the folder `1_af` within the `popgen` folder 

```bash
cd popgen 
cd 1_af
```

### Overall allele frequencies 
Calculate allele frequencies directly form vcf, only for the snps, all samples 

``` bash
plink2 --vcf ../data/chr15.pan.fa.a2fb268.4030258.6a1ecc2.smooth.reliable.vcf.gz \
       --freq \ 
       --allow-extra-chr \
       --vcf-half-call m \
       --chr chm13#chr15 \
       --snps-only \
       --out chr15
```

on a single line 
```bash
plink2 --vcf ../data/chr15.pan.fa.a2fb268.4030258.6a1ecc2.smooth.reliable.vcf.gz  --freq  --allow-extra-chr --vcf-half-call m  --chr chm13#chr15 --snps-only --out chr15

```
This command will produce these files: 

```
-rw-rw-r-- 1 enza enza 33492352 gen 14 11:28 chr15.afreq
-rw-rw-r-- 1 enza enza      923 gen 14 11:28 chr15.log

```
Open the `.log` file  and read it to check that all went well. 

Open the `.afreq` file and look at it 

***

### Population-specific allele frequencies 
Calculate allele frequencies form the plink2 fileset, only for the snps, only for africans and only for americans 

We will use the plink2 filset produced before: 
```
-rw-rw-r-- 1 enza enza        854 gen 13 14:46 chr15.fam
-rw-rw-r-- 1 enza enza        852 gen 13 14:47 chr15.log
-rw-rw-r-- 1 enza enza    9499035 gen 13 14:47 chr15.pgen
-rw-rw-r-- 1 enza enza        503 gen 13 14:47 chr15.psam
-rw-rw-r-- 1 enza enza 1233336557 gen 13 14:47 chr15.pvar

```

The commandline is: 

```bash
plink2 --pfile  ../data/chr15  --keep  ../data/pop/afr.onlyid --freq --allow-extra-chr --snps-only   --out chr15.afr 
```

```bash
plink2 --pfile  ../data/chr15 --keep  ../data/pop/amr.onlyid --freq --allow-extra-chr --snps-only   --out chr15.amr 
```

`--pfile` specifies which plink2 file set to use, you only need to specify the file prefix (e.g. chr15) 

`--keep` filters by individuals in the file 


This will produce: 
```
-rw-rw-r-- 1 enza enza 32483032 gen 14 11:38 chr15.afr.afreq
-rw-rw-r-- 1 enza enza      863 gen 14 11:38 chr15.afr.log
-rw-rw-r-- 1 enza enza 30547659 gen 14 11:38 chr15.amr.afreq
-rw-rw-r-- 1 enza enza      863 gen 14 11:38 chr15.amr.log

```

As usual open the log and check for error messages if any. 
Open the `.afreq` files and look at them 
 
***

### Allele Frequncy Spectrum (AFS) 

In plink2, the `--freq` option includes the `refbin` and `alt1bin` modifiers for generating histogram counts during allele frequency calculations. To use these modifiers, you must specify the bin intervals either directly in the command line or by referencing a file, and using the `refbin-file` and `alt1bin-file` modifiers. 

We can produce the bin info file with a simple shell command that write a sequence form 0.1 to 1 in intervals of 0.05 and store it in a file called `mybins.txt`: 
 
```
LC_NUMERIC="C" seq -s " " 0.1 0.05 1.0 > mybins.txt
````
Look at the `mybins.txt` file: 
```
cat mybins.txt 
```
***
sfs of the alternate allele 

Calculate the counts for the AFS based on alternate allele frequencies:  

```
plink2 --pfile  ../data/chr15 --keep  ../data/pop/afr.onlyid --freq alt1bins-file='mybins.txt'  --allow-extra-chr --snps-only   --out chr15.afr

plink2 --pfile  ../data/chr15 --keep  ../data/pop/amr.onlyid --freq alt1bins-file='mybins.txt'  --allow-extra-chr --snps-only   --out chr15.amr
```

Plot the AFS 

```
Rscript  ../rscripts/plotafs.R chr15.afr.afreq.ref.bins chr15.amr.afreq.ref.bins afr-amr-ALT 
```

<!----
***
sfs of the reference allele 
```
plink2 --pfile  ../data/chr15 --keep  ../data/pop/afr.onlyid --freq refbins-file='mybins.txt'  --allow-extra-chr --snps-only   --out chr15.afr
```

```
plink2 --pfile  ../data/chr15 --keep  ../data/pop/amr.onlyid --freq refbins-file='mybins.txt'  --allow-extra-chr --snps-only   --out chr15.amr
```

```
Rscript  ../rscripts/plotafs.R chr15.afr.afreq.alt1.bins chr15.amr.afreq.alt1.bins afr-amr-REF
``` 
> 

